tensorflow/lite/kernels/cast_test.cc

```diff
@@ -140,6 +140,50 @@ TEST(CastOpModel, CastComplex64ToComplex64) {
            std::complex<float>(6.0f, 16.0f)}));
 }
 
+TEST(CastOpModel, CastComplex64ToUInt8) {
+  CastOpModel m({TensorType_COMPLEX64, {2, 3}}, {TensorType_UINT8, {2, 3}});
+  m.PopulateTensor<std::complex<float>>(
+      m.input(),
+      {std::complex<float>(1.0f, 11.0f), std::complex<float>(2.0f, 12.0f),
+       std::complex<float>(3.0f, 13.0f), std::complex<float>(4.0f, 14.0f),
+       std::complex<float>(5.0f, 15.0f), std::complex<float>(6.0f, 16.0f)});
+  m.Invoke();
+  EXPECT_THAT(m.ExtractVector<uint8_t>(m.output()),
+              ElementsAreArray({1, 2, 3, 4, 5, 6}));
+}
+
+TEST(CastOpModel, CastFloatToUInt8) {
+  CastOpModel m({TensorType_FLOAT32, {3, 2}}, {TensorType_UINT8, {3, 2}});
+  m.PopulateTensor<float>(m.input(), {100.f, 20.f, 3.f, 0.4f, 0.999f, 1.1f});
+  m.Invoke();
+  EXPECT_THAT(m.ExtractVector<uint8_t>(m.output()),
+              ElementsAreArray({100, 20, 3, 0, 0, 1}));
+}
+
+TEST(CastOpModel, CastUInt8ToBool) {
+  CastOpModel m({TensorType_UINT8, {3, 2}}, {TensorType_BOOL, {3, 2}});
+  m.PopulateTensor<uint8_t>(m.input(), {100, 1, 0, 4, 9, 1});
+  m.Invoke();
+  EXPECT_THAT(m.ExtractVector<bool>(m.output()),
+              ElementsAreArray({true, true, false, true, true, true}));
+}
+
+TEST(CastOpModel, CastIntToUInt8) {
+  CastOpModel m({TensorType_INT64, {2, 3}}, {TensorType_UINT8, {2, 3}});
```

Comment:
This test is named `CastIntToUInt8` but you use `INT64` type. This would typically be called `long` or `long int`. For the purposes of this test, I would think you could use `TensorType_INT32` and `int32_t`.

Reply:
@talumbau , thanks for your feedback, i have updated the code as per your suggestion. Kindly check.

Regards
Amit